default_env_value/default_env_value_test.go

```diff
@@ -0,0 +1,19 @@
+package default_env_value
+
+import (
+	"os"
+	"testing"
+)
+
+func TestSetDefaultEnvValue(t *testing.T) {
+	platform := SetDefaultEnvValue("ENV", "aws")
+	err := os.Setenv("ENV", "local")
+	if err != nil {
+		t.Fatal(err)
+	}
+	localEnv := os.Getenv("ENV")
+
+	if localEnv == platform {
+		t.Errorf("expected environment variable: ENV, to have the value: %s, got %s", localEnv, platform)
+	}
+}
```

Comment:
how does this test work? (I'm probably missing something sorry).

1. `SetDefaultEnvValue()` is used to read the `ENV` environment variable but default to `"aws"` (unless `ENV` is set somewhere already it would be equal to `"aws"`
2. then you set the `ENV` environment variable to `"local"` and re-read it into `localEnv`
3. then you check the two are different

Maybe it would be clearer to have something like:

```golang
         const Default = "testDefaultValue"

         // when an environment variable is not set use default
         os.Unsetenv("TEST_ENV")
	 value := SetDefaultEnvValue("TEST_ENV", Default)
         if value != Default {
		t.Errorf(`expected value to be "%s" (default), got "%s"`, Default, value)
         }

         // when an environment variable is set use return its value
         os.Setenv("TEST_ENV", "test_value")
	value := SetDefaultEnvValue("TEST_ENV", Default)
         if value != "test_value" {
		t.Errorf(`expected value to be "%s" (environment variable value), got "%s"`, "test_value", value)
         }
```

or something like that (code not tested)

Reply:
My main point is that AFAICT the test doesn't test that the default is returned when the environment variable is not set.